awx_collection/tests/integration/targets/tower_schedule_rrule/tasks/main.yml

```diff
@@ -0,0 +1,252 @@
+---
+- name: Test too many params
+  debug:
+    msg: "{{ query('awx.awx.tower_schedule_rrule', 'none', 'weekly', start_date='2020-4-16 03:45:07') }}"
+  ignore_errors: true
```

Comment:
looks like these are all error cases, I would like to see something that tests that it produces something usable by the API. I don't see any argument against combining some happy-path values with the `tower_schedule` module, like what we do in the examples. I know the targets are named so they are separated by modules, but other tests already break that pattern (using other modules in tests for some module), and I say you are free to do so as well.

Reply:
Other than firing off a bunch of random output combinations, any ideas on how to do positive testing? I did run into some things where the rrule library was not quite "up to snuff" from a Tower perspective. i.e. the lib does not support TZ so I had to do a manual regex replace to get the TZ in there. Also, Tower demands a INTERVAL in the rrule but the library does not add INTERVAL=1 by default so I had to do a concat if interval == 1.

Reply:
Do you mean you had to add it here in the lookup plugin itself?

https://github.com/ansible/awx/pull/6733/files#diff-5a73bbe4e04087fee21efbcafe352d8eR230

Ideally the library would do most of this, but I'm worried that you may be getting into a hard problem unexpectedly. @ryanpetrello is the whizz on this.